### PR TITLE
Added release activation TRPC calls binding to existing backend service

### PR DESCRIPTION
**Pull request by andrewpatto**
Added TRPC

### CODE REVIEW

File: application/backend/src/api/routes/internal-trpc/release-activation-router.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  deactivate: internalProcedure
+    .input(
+      z.object({
+        releaseKey: z.string(),
+      })
+    )
+    .mutation(async ({ input, ctx }) => {
+      await ctx.releaseService.deactivateRelease(ctx.user, input.releaseKey);
+    }),
+});
```
Comment (andrewpatto) [inline application/backend/src/api/routes/internal-trpc/release-activation-router.ts:26]: the backend for the RPC is quite neat and compact - including in place schemas (or we can externally define the zod schemas)

File: application/backend/src/api/routes/trpc-bootstrap.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  ctx.req.log.trace(authedUser, `isCookieSessionAuthed: user details`);
+
+  return next({
+    ctx: {
+      user: authedUser,
+      ...ctx,
+    },
+  });
+});
```
Comment (andrewpatto) [inline application/backend/src/api/routes/trpc-bootstrap.ts:43]: the middleware passes through strongly typed objects which is nice

File: application/frontend/src/pages/releases/detail/information-box.tsx [hunk 3]
```diff
@@ -48,17 +35,39 @@ export const InformationBox: React.FC<Props> = ({
...
   );
 
+  const queryClient = useQueryClient();
+
+  const activateMutation = trpc.releaseActivation.activate.useMutation();
+  const deactivateMutation = trpc.releaseActivation.deactivate.useMutation();
+
+  // some handy state booleans
+  const mutationInProgress =
+    activateMutation.isLoading || deactivateMutation.isLoading;
```
Comment (andrewpatto) [inline application/frontend/src/pages/releases/detail/information-box.tsx:45]: the frontend includes all the react-query goodness - but is also driven directly off the backend typing

**Review by DoxasticFox — APPROVED** [review]
Nice

**Review by mmalenic — APPROVED** [review]
Cool 😎 So would this remove some of the requirements for common types in elsa-types?

**Comment by andrewpatto** [comment]
> Cool 😎 So would this remove some of the requirements for common types in elsa-types?

Yep - common types would start to disappear

**Pull request merged**